application/constants/templates.py

```diff
@@ -43,3 +49,51 @@ class HookTypes(StrEnum):
     Template application hook types.
     """
     kubernetes_job = 'kubernetes_job'
+
+
+class http(StrEnum):
+    """
+    Template HTTP component types.
+    """
+    url = 'url'
+    method = 'method'
+    headers = 'headers'
+    parameters = 'parameters'
+
+class httpMethods(StrEnum):
+    """
+    Template HTTP component methods.
+    """
+    get = 'GET'
+    post = 'POST'
+    put = 'PUT'
+    patch = 'PATCH'
+    delete = 'DELETE'
+
+
+class httpDeploy(BaseModel):
+    """
+    Deploy schema for HTTP type components.
+    """
+    url: constr(min_length=1, strip_whitespace=True) = Field(description='URL for HTTP deployment', example='http://example.com')
+    method: Union[httpMethods, constr(min_length=1, strip_whitespace=True)] = Field(default=httpMethods.get, description='HTTP method for deployment', example=httpMethods.post)
+    headers: Optional[dict] = Field(description='Headers for HTTP deployment', example={'Content-Type': 'application/json'})
+    parameters: Optional[dict | list[dict]] = Field(description='Parameters for HTTP deployment', example=[{'param1': 'value1'}])
+
+class httpStatus(BaseModel):
```

Comment:
I think I understood what you tried to achieve. You tried to create validation schema for each request. The only distinguish is `default=httpMethods.<method>` for each case. Please investigate next example https://github.com/JovianX/Service-Hub/blob/main/application/services/kubernetes/schemas.py Main trick does:
```
Entities = Annotated[ClusterRoleEntity |
                     <...>
                     ServiceAccountEntity |
                     StatefulSetEntity,
                     Field(discriminator='kind')]
```